app/code/Magento/GoogleGtag/Block/Ga.php

```diff
@@ -0,0 +1,213 @@
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Get a specific page name (may be customized via layout)
+     *
+     * @return string|null
+     */
+    public function getPageName(): ?string
+    {
+        return $this->_getData('page_name');
+    }
+
+    /**
+     * Render GA tracking scripts
+     *
+     * @return string
+     */
+    protected function _toHtml()
+    {
+        if (!$this->googleGtagConfig->isGoogleAnalyticsAvailable()) {
+            return '';
+        }
+
+        return parent::_toHtml();
+    }
+
+    /**
+     * Return cookie restriction mode value.
+     *
+     * @return bool
+     */
+    public function isCookieRestrictionModeEnabled(): bool
+    {
+        return (bool) $this->cookieHelper->isCookieRestrictionModeEnabled();
+    }
+
+    /**
+     * Return current website id.
+     *
+     * @return int
+     */
+    public function getCurrentWebsiteId(): int
+    {
+        return (int) $this->_storeManager->getWebsite()->getId();
+    }
+
+    /**
+     * Return information about page for GA tracking
+     *
+     * @link https://developers.google.com/analytics/devguides/collection/gtagjs
+     * @link https://developers.google.com/analytics/devguides/collection/ga4
+     *
+     * @param string $measurementId
+     * @return array
+     */
+    public function getPageTrackingData($measurementId): array
+    {
+        return [
+            'optPageUrl' => $this->getOptPageUrl(),
+            'measurementId' => $this->escapeHtmlAttr($measurementId, false)
+        ];
+    }
+
+    /**
+     * Return information about order and items for GA tracking.
+     *
+     * @link https://developers.google.com/analytics/devguides/collection/ga4/ecommerce#purchase
+     * @link https://developers.google.com/gtagjs/reference/ga4-events#purchase
+     * @link https://developers.google.com/analytics/devguides/collection/gtagjs/enhanced-ecommerce#product-data
+     * @link https://developers.google.com/gtagjs/reference/event#purchase
+     *
+     * @return array
+     */
+    public function getOrdersTrackingData(): array
+    {
+        $result = [];
+        $orderIds = $this->getOrderIds();
+        if (empty($orderIds) || !is_array($orderIds)) {
+            return $result;
+        }
+        $this->searchCriteriaBuilder->addFilter(
+            'entity_id',
+            $orderIds,
+            'in'
+        );
+        $collection = $this->orderRepository->getList($this->searchCriteriaBuilder->create());
+
+        foreach ($collection->getItems() as $order) {
+            foreach ($order->getAllVisibleItems() as $item) {
+                $result['products'][] = [
+                    'item_id' => $this->escapeJsQuote($item->getSku()),
+                    'item_name' =>  $this->escapeJsQuote($item->getName()),
+                    'price' => number_format((float) $item->getPrice(), 2),
```

Comment:
Why is it exactly 2? Is it required by google api? If not – we have to pull price formats from Magento configuration